### PR TITLE
Finished implementing base of the project

### PR DESCRIPTION
**Pull request by MahdiyarGHD**

### CODE REVIEW

File: src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/Requests/UserAddRequestContract.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace EasyMicroservices.AuthenticationsMicroservice.Contracts.Requests
+{
+    public class UserAddRequestContract : UserContract
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/Requests/UserAddRequestContract.cs]: rename is to AddUserRequestContract

File: src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/Common/UserAuthInputContract.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EasyMicroservices.AuthenticationsMicroservice.Contracts.Common
+{
+    public class UserAuthInputContract
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/Common/UserAuthInputContract.cs]: We are not suing summary names like Auth.
and we have a request and responses folder.
UserContract

File: src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/UserContract.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Threading.Tasks;
+
+namespace EasyMicroservices.AuthenticationsMicroservice.Contracts
+{
+    public class UserContract 
+    {
+        public long Id { get; set; }
+        public string UserName { get; set; }
+        public string Password { get; set; }
+        public string UniqueIdentity { get; set; }
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.AuthenticationsMicroservice.Domain/Contracts/UserContract.cs]: it needs to inheritance from IUniqueIdentitySchema

File: src/CSharp/EasyMicroservices.AuthenticationsMicroservice.WebApi/Controllers/UserController.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        private readonly IJWTManager _jwtManager;
+
+        public UsersController(IContractLogic<UserEntity, UserAddRequestContract, UserUpdateRequest, UserContract, long> contractLogic, IJWTManager jwtManager) : base(contractLogic)
+        {
+            _contractLogic = contractLogic;
+            _jwtManager = jwtManager;
+        }
+
+        [ApiExplorerSettings(IgnoreApi = false)]
+        [HttpPost("Register")]
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.AuthenticationsMicroservice.WebApi/Controllers/UserController.cs]: Please don't use the name in the attribute.
[HttpPost] is enough

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]

**Review by Ali-YousefiTelori — APPROVED** [review]

**Review by Elyas-Amini — APPROVED** [review]

**Pull request merged**